### PR TITLE
AT Version Table ID update

### PR DESCRIPTION
**Pull request by evmiguel**
This PR changes the AtVersion table. It contains migrations to add columns and convert the composite primary key into a serial id primary key. The PR also addresses the affected files on the backend.

### CODE REVIEW

File: server/migrations/20220427204420-changeAtVersionColumn.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+'use strict';
```
Comment (evmiguel) [inline server/migrations/20220427204420-changeAtVersionColumn.js:1]: @howard-e @alflennik This is the last migration file. For some reason changing the column name from `atVersion` to `name` isn't working. Trying to figure this one out, but could use some help

Comment (howard-e) [reply server/migrations/20220427204420-changeAtVersionColumn.js:1]: What is the error that's thrown?

You could try removing the `atVersion` column then adding the `name` column in the migration if all else fails

Comment (howard-e) [reply server/migrations/20220427204420-changeAtVersionColumn.js:1]: Okay seems that's solved! 👍🏽

File: server/migrations/20220427204420-changeAtVersionColumn.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    },
+
+    down: async queryInterface => {
+        queryInterface.sequelize.transaction(async transaction => {
+            await queryInterface.sequelize.query(
+                'ALTER TABLE public."AtVersion" RENAME COLUMN "name" TO "atVersion";',
+                { transaction }
+            );
+        });
+    }
```
Comment (alflennik) [inline server/migrations/20220427204420-changeAtVersionColumn.js:20]: It should be possible to combine all four migrations into one, that'll just make the migration more compact and easier to take in all at once.

Comment (alflennik) [reply server/migrations/20220427204420-changeAtVersionColumn.js:20]: Does the migration work? Does the ID field auto-populate? If so, that's one of the biggest hurdles overcome, which would be sweet.

Comment (alflennik) [reply server/migrations/20220427204420-changeAtVersionColumn.js:20]: I think we should also make a symmetrical set of changes to the Browser table, i.e. giving it an ID field and possibly a releasedAt field too. @s3ththompson are we going to track the dates where browser versions were released as well?

Comment (evmiguel) [reply server/migrations/20220427204420-changeAtVersionColumn.js:20]: > Does the migration work? Does the ID field auto-populate? If so, that's one of the biggest hurdles overcome, which would be sweet.

Yes, the IDs are auto-populated.

Comment (evmiguel) [reply server/migrations/20220427204420-changeAtVersionColumn.js:20]: > It should be possible to combine all four migrations into one, that'll just make the migration more compact and easier to take in all at once.

I was running into some issues with the ordering of changes, which is why I broke them out. I know multiple files isn't ideal, but they are easier to read, if anything.

File: server/migrations/20220427192638-changeConstraintsAtVersion.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                'AtVersion',
+                'AtVersion_pkey',
+                { transaction }
+            );
+        });
+    },
+
+    down: queryInterface => {
+        return queryInterface.sequelize.transaction(async transaction => {
+            await queryInterface.addConstraint(
```
Comment (howard-e) [inline server/migrations/20220427192638-changeConstraintsAtVersion.js:16]: Seems this should be changed from the [v6](https://sequelize.org/api/v6/class/src/dialects/abstract/query-interface.js~queryinterface#instance-method-addConstraint) version to the [v5](https://sequelize.org/v5/class/lib/query-interface.js~queryinterface#instance-method-addConstraint) as well

Comment (evmiguel) [reply server/migrations/20220427192638-changeConstraintsAtVersion.js:16]: Good catch!

**Review by howard-e — APPROVED** [review]

**Review by alflennik — APPROVED** [review]

**Pull request merged**